examples/notebooks/03_model_scoring.py

```diff
@@ -0,0 +1,220 @@
+Metadata for a 34 channel H5 files described above is given below.
+"""
+# %%
+from earth2mip.initial_conditions import hdf5
+
+metadata = {
+    "h5_path": "fields",
+    "attrs": {"decription": "Custom HDF5 data"},
+    "dims": ["time", "channel", "lat", "lon"],
+    "coords": {
+        "lat": np.linspace(90, -90, 721),
+        "lon": np.linspace(0, 359.75, 1440),
+        "channel": [
+            "u10m",
+            "v10m",
+            "t2m",
+            "sp",
+            "msl",
+            "t850",
+            "u1000",
+            "v1000",
+            "z1000",
+            "u850",
+            "v850",
+            "z850",
+            "u500",
+            "v500",
+            "z500",
+            "t500",
+            "z50",
+            "r500",
+            "r850",
+            "tcwv",
+            "u100m",
+            "v100m",
+            "u250",
+            "v250",
+            "z250",
+            "t250",
+            "u100",
+            "v100",
+            "z100",
+            "t100",
+            "u900",
+            "v900",
+            "z900",
+            "t900",
+        ],
+    },
+}
+
+datasource = hdf5.DataSource(root=h5_folder, metadata=metadata)
+
+# Test to see if our datasource is working
+time = datetime.datetime(2017, 5, 1, 18)
+out = datasource[time]
+
+# %% [markdown]
+"""
+## Loading Models
+
+With the HDF5 datasource loaded the next step is to load our model we wish to score.
+In this tutorial we will be using the built in FourcastNet model.
+Take note of the `e2mip://` which will direct Earth-2 MIP to load a known model package.
+FourcastNet is selected here simply because its a 34 channel model which aligns with the
+H5 files described above.
+"""
+# %%
+import earth2mip.networks.fcn as fcn
+from earth2mip import registry
+from modulus.distributed import DistributedManager
+
+device = DistributedManager().device
+package = registry.get_model("e2mip://fcn")
+model = fcn.load(package, device=device)
+
+# %% [markdown]
+"""
+## Running Scoring
+
+With the datasource and model loaded, scoring can now be performed.
+To score this we will run 10 day forecasts over the span of the entire year at 30 day
+intervals.
+For research, one would typically want this to be much more comprehensive so feel free
+to customize for you're use case.
+
+The `score_deterministic` API provides a simple way to calculate RMSE and ACC scores.
+ACC scores require climatology which is beyond the scope of this example, thus zero
+values will be provided and only the RMSE will be of concern.
+"""
+
+# %%
+from earth2mip.inference_medium_range import score_deterministic
+
+time = datetime.datetime(2017, 1, 1, 0)
+initial_times = [time + datetime.timedelta(days=30 * i) for i in range(12)]
+
+output = score_deterministic(
+    model,
+    n=40,  # 6 hour timesteps
+    initial_times=initial_times,
+    data_source=datasource,
+    time_mean=np.zeros((34, 721, 1440)),
+)
+output.to_netcdf("scoring_output.nc")
+print(output)
```

Comment:
I assume this was a caching step. To make the notebook idempotent (run twice same answer) but also fast you could do something like this (and also remove the cleanup step).


```suggestion
if not os.path.exists("scoring_output.nc"):
    output = score_deterministic(
        model,
        n=40,  # 6 hour timesteps
        initial_times=initial_times,
        data_source=datasource,
        time_mean=np.zeros((34, 721, 1440)),
    )
    output.to_netcdf("scoring_output.nc")
    
    
dataset = xarray.open_dataset("scoring_output.nc")
print(dataset)
```

Reply:
Exactly, thanks for the catch.